tools/code-owners-parser/CodeOwnersParser/MatchedCodeownersEntry.cs

```diff
@@ -148,31 +171,34 @@ private Regex ConvertToRegex(string targetPath, string codeownersPath)
                     $"\"{DoubleStar}\" or \"{SingleStar}\"");
             }
 
-            pattern = pattern.Replace("**", DoubleStar);
+            // We replace "/**/", not "**", because we disallow "**" in any other context.
```

Comment:
Note to self: make a pass over all the comments and PR description to ensure they are up to date. The validation rules have changed, we no longer treat target path `/a` as potentially dir.

We also forbid `/**/` if it is a suffix.